WordPress/src/main/java/org/wordpress/android/ui/accounts/LoginMagicLinkInterceptActivity.java

```diff
@@ -4,19 +4,23 @@
 import android.content.Intent;
 import android.os.Bundle;
 
-import org.wordpress.android.analytics.AnalyticsTracker;
+import org.wordpress.android.login.LoginAnalyticsListener;
 import org.wordpress.android.ui.main.WPMainActivity;
 
+import javax.inject.Inject;
+
 /**
  * Deep link receiver for magic links. Starts {@link WPMainActivity} where flow is routed to login
  * or signup based on deep link scheme, host, and parameters.
  */
 public class LoginMagicLinkInterceptActivity extends Activity {
+    @Inject protected LoginAnalyticsListener mAnalyticsListener;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
 
-        AnalyticsTracker.track(AnalyticsTracker.Stat.LOGIN_MAGIC_LINK_OPENED);
+         mAnalyticsListener.trackLoginMagicLinkOpened();
```

Comment:
I think we need to inject the `mAnalyticsListener` instance before using it, otherwise it will be `null`.

Reply:
👍 yeah, this and `MagicLinkRequestFragment` below will both need the `((WordPress) getActivity().getApplication()).component().inject(this)`/`((WordPress) getApplication()).component().inject(this)` calls as usual, and to be added to the `AppComponent`.

Reply:
I added the inject statements in 48ac685 and 00117ef.

Reply:
Just to be sure, is the injection happening before the call to onCreate()'s super? I know that sometimes we do place it like that but I'm not sure when :)

Reply:
Yep, according to the [Dagger documentation](https://google.github.io/dagger/android.html), it states:
> Finally, in your `Activity.onCreate()` method, call `AndroidInjection.inject(this)` _before_ calling `super.onCreate();`:

Reply:
We're having the injection happening before the super call for `AndroidInjection.inject()` elsewhere, yeap. Does that apply for the `.component().inject()` calls too though? I'm under the impression that we call the latter after the super. Any insight on this one @aforcier ?

Reply:
Yes, those docs apply to the `AndroidInjection` method of Dagger `2.11` - we're only using this for login library fragments at the moment. Everywhere that uses the old `component.inject()` form should follow the convention elsewhere in the WordPress module, just to be consistent/prevent any surprises.

Reply:
Do we want to do the same for `MagicLinkRequestFragment` too?

Reply:
I moved the statement in 2d9d592259b70f3ad31455c3c4951ab0c17447ba.